core/src/main/java/org/apache/accumulo/core/spi/crypto/AESCryptoService.java

```diff
@@ -211,29 +222,27 @@ public void setEncFek(byte[] encFek) {
 
   }
 
-  private static byte[] createCryptoParameters(String version, Key encryptingKek,
-      String encryptingKekId, String encryptingKeyManager, Key fek) {
-
-    byte[] bytes;
-    try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        DataOutputStream params = new DataOutputStream(baos)) {
-      params.writeUTF(AESCryptoService.class.getName());
-      params.writeUTF(version);
-      params.writeUTF(encryptingKeyManager);
-      params.writeUTF(encryptingKekId);
-      byte[] wrappedFek = wrapKey(fek, encryptingKek);
-      params.writeInt(wrappedFek.length);
-      params.write(wrappedFek);
-
-      bytes = baos.toByteArray();
+  private static byte[] createCryptoParameters(final String version, final Key encryptingKek,
+      final String encryptingKekId, final String encryptingKeyManager, final Key fek) {
+
+    final byte[] wrappedFek = wrapKey(fek, encryptingKek);
+    try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+      try (DataOutputStream params = new DataOutputStream(baos)) {
+        params.writeUTF(AESCryptoService.class.getName());
+        params.writeUTF(version);
+        params.writeUTF(encryptingKeyManager);
+        params.writeUTF(encryptingKekId);
+        params.writeInt(wrappedFek.length);
+        params.write(wrappedFek);
+      }
+      return baos.toByteArray();
```

Comment:
This change closes the DataOutputStream to ensure that it's flushed before getting the bytes. I don't think this is an issue now, but the internals of DataOutputStream could change...

Reply:
I am not following you... the DataOutputStream is already being handled as a part of the try-with-resources. Maybe you didn't see it cause its on a second line?

Reply:
I saw it, but the DataOutputStream is closed at the end of the try block. You are calling baos.toByteArray() *before* the end of the try block.

Reply:
OK, I think see what you are doing here. Couldn't we just call `params.flush();` before returning the byte array?

Reply:
Or a call to flush on both streams to be safe?

Reply:
Yes, you could do that too.